src/main/java/teammates/storage/entity/Notification.java

```diff
@@ -75,19 +78,25 @@ public Notification(Instant startTime, Instant endTime, String type, String targ
      * Instantiates a new notification, with all fields passed in as parameters.
      * This is mainly for conversion from attributes to entity.
      */
-    public Notification(String notificationId, Instant startTime, Instant endTime, String type, String targetUser,
-            String title, String message, boolean shown, Instant createdAt, Instant updatedAt) {
+    public Notification(String notificationId, Instant startTime, Instant endTime,
+                        NotificationType type, NotificationTargetUser targetUser,
+                        String title, String message, boolean shown, Instant createdAt, Instant updatedAt) {
         this.setStartTime(startTime);
         this.setEndTime(endTime);
         this.setType(type);
         this.setTargetUser(targetUser);
         this.setTitle(title);
         this.setMessage(message);
-        this.setCreatedAt(createdAt);
+        if (createdAt == null) {
```

Comment:
Can just shorten this to 
`this.createdAt = createdAt == null ? this.setCreatedAt(Instant.now()): this.setCreatedAt(createdAt)`